src/frontend/src/binder/expr/binary_op.rs

```diff
@@ -89,4 +93,20 @@ impl Binder {
         };
         Ok(FunctionCall::new(func_type, vec![left, right])?.into())
     }
+
+    fn bind_regex_match(&mut self, left: ExprImpl, right: ExprImpl) -> Result<ExprImpl> {
+        Ok(FunctionCall::new(
+            ExprType::IsNotNull,
```

Comment:
This rewrite is not equivalent when either `left` or `right` is `null`. The operator returns `null` but this returns `false`. Let's open an issue to track the fix.